mlflow/projects/utils.py

```diff
@@ -83,31 +80,9 @@ def _is_file_uri(uri):
     return _FILE_URI_REGEX.match(uri)
 
 
-def _is_git_repo(path):
-    """Returns True if passed-in path is a valid git repository"""
-    import git
-
-    try:
-        git.Repo(path)
-        return True
-    except git.exc.InvalidGitRepositoryError:
-        return False
-
-
 def _is_local_uri(uri):
     """Returns True if passed-in URI should be interpreted as a path on the local filesystem."""
-    if _GIT_URI_REGEX.match(uri):
-        return False
-
-    parsed_uri = urllib.parse.urlparse(uri)
-    drive = pathlib.Path(uri).drive
-
-    if drive != "" and drive.lower()[0] == parsed_uri.scheme:
-        return not _is_git_repo(uri)
-    elif parsed_uri.scheme in ("file", ""):
-        return not _is_git_repo(parsed_uri.path)
-    else:
-        return False
+    return pathlib.Path(uri).exists() and pathlib.Path(uri).is_dir()
```

Comment:
Does `pathlib.Path.exists` work properly when `uri` is a file URI?

```
>>> Path.cwd()
PosixPath('/home/haru/Desktop/repositories/mlflow')
>>> Path.cwd().as_uri()
'file:///home/haru/Desktop/repositories/mlflow'
>>> Path(Path.cwd().as_uri()).exists()
False
>>> Path(Path.cwd().as_uri(), "setup.py").exists()
False
```

Reply:
I see what you mean, we could check that using the file uri regex, right? 
Let me make a small update.

Reply:
Done, I think this should work regardless of OS and should also work for file URIs.